docs/UserGuide.md

```diff
@@ -159,6 +159,108 @@ Failure Output:
+| **-before**  | Schedule                | Lessons that occur before (not including) the specified date. Valid input formats: `YYYY/MM/DD` or `YY/MM/DD` or `MM/DD` or `DD` | `filter -before 2023/12/01`, `filter -before 23/12/01`, `filter -before 12/01`, `filter -before 1` |
+| **-on**      | Schedule                | Lessons that occur on the specified date. Valid input formats: `YYYY/MM/DD` or `YY/MM/DD` or `MM/DD` or `DD`                     | `filter -on 2023/12/01`, `filter -on 23/12/01`, `filter -on 12/01`, `filter -on 1`                 |
+| **-after**   | Schedule                | Lessons that occur after (not including) the specified date. Valid input formats: `YYYY/MM/DD` or `YY/MM/DD` or `MM/DD` or `DD`  | `filter -after 2023/12/01`, `filter -after 23/12/01`, `filter -after 12/01`, `filter -after 1`     |
+| **-remark**  | Students, Schedule      | Lessons or students whose remarks partially or fully matches the remark entered                                                  | `filter -remark assistance`                                                                        |                                                                                                                                                                                                                                 |
+| **-tag**     | Students                | Students who have at least one tag that fully matches the tag name entered                                                       | `filter -tag primary`                                                                              |
+ 
+Notes
+- When filtering, all text is case-insensitive. That is, `filter -name LEAH` is the same as `filter -name leah`. 
+- You can filter by any number of flags, and in any order.
+- To reset the filter (view all students / lessons), type `list students` or `list schedule`.
+
+Examples:
+- In `SCHEDULE` list:
+  - `filter -name CS2103T -before 2023/12/11`
+- In `STUDENT` list:
+  - `filter -name Alex -tag primary -subject Mathematics`
+
+Success Output:
+- For the command `filter -name Alex -tag primary -subject Mathematics` in the _Student List_ with all detail fields shown
+![Success for filter 1](images/filter/filter_positive_1.png)
+- For the command `filter -name Alex -tag secondary` (No such students matching the filter found)
+![Success for filter 2](images/filter/filter_positive_2.png)
+
+Failure output:
+- When there are invalid values specified in the command `filter -on 2/2/2` (invalid date format)
+![Error for filter](images/filter/filter_negative_1.png)
+```
+Invalid lesson format: 2/2/2 is not a valid date, please use yyyy/mm/dd or mm/dd or dd
+for example, assume today is 2023/11/3, to add 2023/11/29, could use 29, 11/29, 2023/11/29 or 23/11/29. 
+Usage: filter -(any number of unique -[name|subject|before|on|after|remark] [value]). 
+```
```

Comment:
i think jun hiang mentioned that ideally we group these into each student/schedule state

Reply:
Will be resolved when we work on the new unified UG!